packages/functional-tests/tests/subscription-tests/subscription.spec.ts

```diff
@@ -53,3 +53,54 @@ test.describe('subscription test with cc and paypal', () => {
     expect(await relier.isPro()).toBe(true);
   });*/
 });
+
+test.describe('metrics - flow metrics query params', () => {
+  test.beforeEach(() => {
+    test.slow();
+  });
+
+  test('logged in and toggle on Share Data, Checkout to have flow params in URL', async ({
+    pages: { settings, relier, page },
+  }) => {
+    await settings.goto();
+    await settings.dataCollection.toggleShareData('on');
+
+    const ariaChecked = page
+      .locator('[data-testid=metrics-opt-out]')
+      .getAttribute('aria-checked');
+    expect(await ariaChecked).toBe('true');
+
+    await relier.goto();
+    await relier.clickSubscribe();
+    // expect(page.url()).toContain('&flow_begin_time=');
+  });
+
+  test('logged in and toggle off Share Data, Checkout to not have flow params in URL', async ({
+    pages: { settings, relier, page },
+  }) => {
+    await settings.goto();
+    await settings.dataCollection.toggleShareData('off');
+
+    const ariaChecked = page
+      .locator('[data-testid=metrics-opt-out]')
+      .getAttribute('aria-checked');
+    expect(await ariaChecked).toBe('false');
+
+    await relier.goto();
+    await relier.clickSubscribe();
+    expect(page.url()).toBe(
+      'http://localhost:3030/subscriptions/products/prod_GqM9ToKK62qjkK?plan=plan_GqM9N6qyhvxaVk'
```

Comment:
nit: Should these product and plan ids be in `const`s or env vars?

```sh
git grep "prod_GqM9ToKK62qjkK" | wc -l # 33
git grep "plan_GqM9N6qyhvxaVk" | wc -l # 27
```

Reply:
👍 I've updated this to just check if it contains a flow metric query param instead.